scripts/build_gh_pages

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
+# shellcheck source=scripts/common
+source "$SCRIPTS_DIR/common"
+
+readonly SRC_SUBDIRS=(sdk/rust oak/server/rust)
+readonly DOC_SUBDIRS=(sdk server)
+
+# Update the gh-pages branch. Note that `cargo doc` is **not deterministic** so
+# this should only be done when there is a real change.
+readonly BRANCH=${1:-master}
+readonly PREV_COMMIT=$(git log --oneline -n 1 gh-pages | sed 's/.*branch at \([0-9a-f]*\)/\1/')
+
+readonly CHANGES=$(git diff "${PREV_COMMIT}..${BRANCH}" -- "${SRC_SUBDIRS[@]}" | grep -e '[+-]//[/!]')
+
+if [ -z "${CHANGES}" ]; then
+  echo "No doc comment changes found in ${PREV_COMMIT}..${BRANCH}" subdirs "${SRC_SUBDIRS[@]}"
+  exit 0
+fi
+
+git checkout "${BRANCH}"
+readonly BRANCH_SHA1=$(git rev-parse --short HEAD)
```

Comment:
Quote RHS?

Reply:
Why? There are no variables in it.

Reply:
I thought it would do something weird if there are whitespaces or other weird characters, but it may just be my irrational fear of Bash speaking.

Reply:
Nothing irrational there :-)